### PR TITLE
fix: honour MDN requests

### PR DESCRIPTION
**Pull request by miaulalala**
Needs https://github.com/nextcloud/mail/pull/9364

Fixes https://github.com/nextcloud/mail/issues/9365

image

### CODE REVIEW

File: lib/Migration/Version3600Date20240319172810.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2024 FIXME Your name <your@email.com>
```
Comment (ChristophWurst) [inline lib/Migration/Version3600Date20240319172810.php:6]: fixme

**Review by ChristophWurst — COMMENTED** [review]
Code looks good :+1:

File: lib/Migration/Version3700Date20240506161400.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2024 FIXME Your name <your@email.com>
```
Comment (ChristophWurst) [inline lib/Migration/Version3700Date20240506161400.php:6]: yeah

**Review by st3iny — CHANGES_REQUESTED** [review]
Tested and works (mostly).

## Some feedback:
1. Please adjust version in `info.xml` to trigger the migration.
2. MDN state is lost when saving a draft, closing it and editing it again. I'm currently investigating that.
3. MDN state is also lost when canceling a message in the outbox.

**Review by st3iny — APPROVED** [review]
Tested and works. I addressed my own feedback.

File: appinfo/info.xml [hunk 3]
```diff
@@ -29,7 +29,7 @@ The rating depends on the installed text processing backend. See [the rating ove
 
 Learn more about the Nextcloud Ethical AI Rating [in our blog](https://nextcloud.com/blog/nextcloud-ethical-ai-rating/).
 	]]></description>
-	<version>3.7.0-alpha.2</version>
+	<version>3.7.0-alpha.3</version>
```
Comment (miaulalala) [inline appinfo/info.xml:32]: 3.7.0 has already been and gone - need to update to 3.8.0-alpha.x

Comment (st3iny) [reply appinfo/info.xml:32]: Oopsie, could you please fix it when squashing? Thanks :)

**Pull request merged**